demisto_sdk/commands/common/hook_validations/integration.py

```diff
@@ -599,6 +617,96 @@ def _get_invalid_dbot_outputs(context_outputs_paths, context_outputs_description
+            error_message, error_code = Errors.command_reputation_output_is_missing(
+                command.get("name", ""),
+                output,
+                ioc,
+            )
+            if self.handle_error(
+                error_message,
+                error_code,
+                file_path=self.file_path,
+                warning=self.structure_validator.quiet_bc,
+            ):
+                return False
+
+        return True
+
+    @staticmethod
+    def extract_object_name(context_path):
+        for object_name in MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES:
+            if object_name.lower() in context_path.lower():
+                return object_name
+        return None
+
+    def validate_spelling(self, command: dict, outputs_results: OutputsResults) -> bool:
+        """Validates command ouputs listed MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES
+           are spelled and capitalized correctly.
+        Returns:
+            True if command outputs are spelled right.
+        """
+        for output in command.get("outputs") or []:
+            context_path = output.get("contextPath", "")
+            outputs_results.add_path(context_path)
+
+            if reputation_object := self.extract_object_name(context_path):
+                outputs_results.add_reputation_object(reputation_object)
+                if (
+                    reputation_object not in context_path
+                ):  # the output is not spelled as expected
+                    (
+                        error_message,
+                        error_code,
+                    ) = Errors.command_reputation_output_capitalization_incorrect(
+                        command.get("name"), context_path, reputation_object
+                    )
+                    if self.handle_error(
+                        error_message,
+                        error_code,
+                        file_path=self.file_path,
+                        warning=self.structure_validator.quiet_bc,
+                    ):
+                        return False
+        return True
+
+    @error_codes("IN158")
+    def validate_command_outputs(self) -> bool:
+        """Check if a command is using the following objects: file, ip, endpoint, email, domain, url, cve,
+        infofile, certificate, if so, validate it is according to context
+        standards: https://xsoar.pan.dev/docs/integrations/context-standards
+        Returns:
+            bool.
+            Whether a command is using the mentioned objects and their outputs
+            according to the context standards
+        """
+        commands = self.current_file.get("script", {}).get("commands", [])
+        output_results = OutputsResults()
+        for command in commands:
+            if not (
+                self.validate_spelling(command, output_results)
+                and self.validate_all_reputation_outputs_exist(command, output_results)
+            ):
+                return False
```

Comment:
This will stop after just one bad command, causing users frustration as they will fix one and suddenly another will show as faulty. Keep a boolean `result` variable as we discussed yesterday, and return it after checking all commands.

Reply:
done.